### PR TITLE
Kernel: Add MSI support for PCI devices

### PR DESCRIPTION
**Pull request by Panky-codes**
Adds MSI support for PCI devices. This is a follow-up to #18580 which added MSIx support.

Extended AHCI to use MSI instead of pin-based interrupts.

Test:
```
SERENITY_RUN=q35 Meta/serenity.sh run
```
Output from `lsirq`:
image

### CODE REVIEW

**Comment by supercomputer7** [comment]
I will give this a look today or tomorrow. Thank you for doing this!

**Comment by supercomputer7** [comment]
Please change the second commit message:
` Kernel: Use PCIIdentifier is_msi_capable API to retrieve MSI status` => ` Kernel: Use PCIDeviceIdentifier is_msi_capable API to retrieve MSI status` 
`is_msi_capable() API from the PCIIdentifier class that belongs to the` => `is_msi_capable() API from the PCIDeviceIdentifier class that belongs to the`

**Review by supercomputer7 — APPROVED** [review]
Seems really good. Just fix the commit message as I said before, but otherwise I think this is merge-able :)

File: Kernel/Storage/ATA/AHCI/Controller.cpp [hunk 1]
```diff
@@ -167,12 +167,13 @@ UNMAP_AFTER_INIT ErrorOr<void> AHCIController::initialize_hba(PCI::DeviceIdentif
     u32 version = hba().control_regs.version;
 
     hba().control_regs.ghc = 0x80000000; // Ensure that HBA knows we are AHCI aware.
-    PCI::enable_interrupt_line(device_identifier());
     PCI::enable_bus_mastering(device_identifier());
+    TRY(reserve_irqs(1, true));
+    auto irq = TRY(allocate_irq(0));
```
Comment (GMTA) [inline Kernel/Storage/ATA/AHCI/Controller.cpp:172]: This `TRY` can probably be a `MUST` if `reserve_irqs` succeeded earlier?

Comment (Panky-codes) [reply Kernel/Storage/ATA/AHCI/Controller.cpp:172]: Yeah, that makes sense. I will change that!

**Comment by Panky-codes** [comment]
ping @GMTA :)

**Review by GMTA — APPROVED** [review]

**Pull request merged**